std/regex/internal/ir.d

```diff
@@ -495,6 +615,35 @@ package(std.regex):
     public const(BitTable)[] filters;      // bloom filters for conditional loops
     uint[] backrefed;                      // bit array of backreferenced submatches
     Kickstart!Char kickstart;
+    MatcherFactory!Char factory;           // produces optimal matcher for this pattern
+
+    const(Regex) withFactory(MatcherFactory!Char factory) pure const @trusted
+    {
+        auto r = cast() this;
+        r.factory = factory;
+        return r;
```

Comment:
While this is a nice pattern, you completely work around `const` here.
At least it's internal ;-)

Reply:
Yes. I feel like a compiler can prove this is legal iff all member of struct are value types or const references.